src/doc/src/reference/breaking.md

```diff
@@ -0,0 +1,536 @@
+3. **Patch**: incremented for backwards-compatible bug fixes.
+
+> **Note** As well as the rule that versions **<1.0.0** can change anything.
+
+Typically, a **breaking change** is a change that, *strictly speaking*, can cause downstream
+code to fail to compile
+
+In Rust today, almost any change is technically a
+breaking change. For example, given the way that globs currently work, *adding
+any public item* to a library can break its clients (more on that later). But
+not all breaking changes are equal. This has led to Rust's ecosystem adopting the following:
+
+**All major changes are breaking, but not all breaking
+changes are major.**
+
+> **Note**: This belief was codified for the standard library in [RFC 1105][rfc]
+
+## Principles of the policy
+
+The basic policy is that **the same code should be able to run
+against different minor revisions**. Furthermore, minor changes should require
+at most a few local *annotations* to the code you are developing, and in
+principle no changes to your dependencies.
+
+In more detail:
+
+* Minor changes should require at most minor amounts of work upon upgrade. For
+  example, changes that may require occasional type annotations or use of [UFCS][ufcs]
+  to disambiguate are not automatically "major" changes. (But in such cases, one
+  must evaluate how widespread these "minor" changes are).
+
+* In principle, it should be possible to produce a version of dependency code
+  that *will not break* when upgrading other dependencies, or Rust itself, to a
+  new minor revision. This goes hand-in-hand with the above bullet; as we will
+  see, it's possible to save a fully "elaborated" version of upstream code that
+  does not require any disambiguation. The "in principle" refers to the fact
+  that getting there may require some additional tooling or language support,
+  which this RFC outlines.
+
+That means that any breakage in a minor release must be very "shallow": it must
+always be possible to locally fix the problem through some kind of
+disambiguation *that could have been done in advance* (by using more explicit
+forms) or other annotation (like disabling a lint). It means that minor changes in upstream dependencies
+can never leave a downstream dependency in a state that requires breaking changes to their own code.
+
+**Although this general policy allows some (very limited) breakage in minor
+releases, it is not a license to make these changes blindly**. The breakage that
+this policy permits, aside from being very simple to fix, are also unlikely to occur
+often in practice.
+
+## Policy by language feature
+
+Most of the policy is simplest to lay out with reference to specific language
+features and the way that APIs using them can, and cannot, evolve in a minor
+release.
+
+**Breaking changes are assumed to be major changes unless otherwise stated**.
+The RFC covers many, but not all breaking changes that are major; it covers
+*all* breaking changes that are considered minor.
```

Comment:
This is a bit of a strong statement, I'm not sure if it is accurate since I don't consider this list to be exhaustive.